client/components/log-in-form.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+
+export default class LogInForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      validated: false,
+      usernameError: '',
+      passwordError: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.name === 'username') {
+      this.setState({
+        username: event.target.value
+      });
+    } else {
+      this.setState({
+        password: event.target.value
+      });
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const form = event.currentTarget;
+
+    const { username, password } = this.state;
+
+    const data = {
+      username,
+      password
+    };
+
+    fetch('/api/auth/log-in', {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json'
+      },
+      body: JSON.stringify(data)
+    })
+      .then(response => {})
+      .catch(err => console.error(err));
+    form.reset();
```

Comment:
You don't want the form to reset until the request has finished. So the `reset` method call should be in your `.then` callback function